cl/donate/api_views.py

```diff
@@ -0,0 +1,215 @@
+import json
+from http import HTTPStatus
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpResponse
+from rest_framework import mixins, serializers, viewsets
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from cl.donate.api_permissions import AllowNeonWebhook
+from cl.donate.models import NeonMembership, NeonWebhookEvents
+from cl.lib.neon_utils import NeonClient
+
+
+class NeonMembershipWebhookSerializer(serializers.Serializer):
+    eventTrigger = serializers.CharField(required=True)
+    eventTimestamp = serializers.DateTimeField(required=True)
+    data = serializers.DictField()
+
+
+class MembershipWebhookViewSet(
+    mixins.CreateModelMixin, viewsets.GenericViewSet
+):
+    permission_classes = [AllowNeonWebhook]
+    serializer_class = NeonMembershipWebhookSerializer
+    queryset = NeonMembership.objects.all()
+
+    def create(self, request: Request, *args, **kwargs):
+        """
+        Processes membership webhooks received from Neon CRM.
+
+        This method handles POST requests containing JSON data from Neon CRM.
+        It extracts relevant information from the data and updates the
+        corresponding membership record in the database.
+
+        Args:
+            request (Request): A request object containing the webhook data.
+
+        Returns:
+            Response: A HttpResponse object with status code 200 if successful,
+            or another appropriate response if an error occurs.
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        webhook_data = serializer.validated_data
+        self._store_webhook_payload(webhook_data)
+        match webhook_data["eventTrigger"]:
+            case "createMembership" | "editMembership":
+                self._handle_membership_creation_or_update(webhook_data)
+            case "deleteMembership":
+                self._handle_membership_deletion(webhook_data)
+            case "updateMembership":
+                pass
+            case _:
+                trigger = webhook_data["eventTrigger"]
+                raise NotImplementedError(f"Unknown event trigger-{trigger}")
+
+        return Response(status=HTTPStatus.CREATED)
+
+    def _get_member_record(self, account_id: str) -> User:
+        """
+        Retrieves a user record associated with a Neon CRM account ID.
+
+        This method attempts to find a matching Django user in two ways:
+
+        1. first tries to directly query the database for a user whose
+        `neon_account_id` field matches the provided `account_id`.
+        2. If no matching user is found in the database, it fetches the
+        account email address from the Neon API using the `account_id`.
+        It then tries to find a user whose email address matches the
+        retrieved Neon account email.
```

Comment:
I think if we still can't find such an account, we should create a stub. There's an example of this when anonymous users create donations. 

And later, when the account is created, we should convert the stub to the real thing, such that the donations for the user (or their membership) is ready and waiting for them. The idea is that you should be able to donate first, and then create an account second.

Reply:
> I think if we still can't find such an account, we should create a stub.

Got it! I'll use the helper method to create stubs.